mesh/mesh.hpp

```diff
@@ -409,8 +409,14 @@ class Mesh
    /// Refine a mixed 3D mesh uniformly.
    virtual void UniformRefinement3D() { UniformRefinement3D_base(); }
 
-   /// Refine NURBS mesh.
-   virtual void NURBSUniformRefinement();
+   /** @brief Refine NURBS mesh, with an optional refinement factor.
+
+       @param[in] rf  Optional refinement factor. If scalar, the factor is used
+                      for all dimensions. If an array, factors can be specified
+                      for each dimension.
+       @param[in] tol NURBS geometry deviation tolerance. */
+   virtual void NURBSUniformRefinement(int rf = 2, double tol = 1.0e-12);
```

Comment:
This tolerance parameter is unfortunately tricky, and I simply added a reference to Algorithm A5.8 of the NURBS book. That book has some discussion of the tolerance, which is a little complicated and would be difficult to reproduce in the code comments, so I hope a reference is sufficient. 

So far, in my testing, I have not had to use a value other than the default 1e-12. I expect that users can ignore this parameter, unless a particular mesh has problems. I suppose if the parameter needs to be changed, it will be done by trial and error, rather than by theoretical understanding and calculation of the parameter. In other words, I don't fully understand it yet and would just try different values until it works and the mesh looks good.

Reply:
Sounds good! Lets keep it as it is now. It is a bit unfortunate that the tolerance is an input specific for NURBS in a function that is applicable for many types of mesh in `Mesh::UniformRefinement`. An alternative to think of would be to remove the input option `tol` in `Mesh::UniformRefinement`  and move `mesh::NURBSUniformRefinement` to public. In this way specific tolerances can be set using the dedicated NURBS refinement function and `Mesh::UniformRefinement` has one input option less.

Reply:
I agree. Before this PR, `UniformRefinement` only called `NURBSUniformRefinement` in the NURBS case. Now, with more inputs specific to the NURBS case, I think it makes more sense to call `NURBSUniformRefinement` directly instead, and verify that it is a NURBS mesh. I made this change in 041ce914d870a61d7d1e5c565c59fa351bde02c5

Reply:
Nice!!